### ISSUE TITLE
TwoDimensionalTest.testEvolvingConstraintsInterior fails

### ISSUE THREAD

**Issue by snorfalorpagus** [bradfordboyle/pyglpk#2]
Currently the `TwoDimensionalTest.testEvolvingConstraintsInterior` test fails against GLPK 4.55 (and perhaps older - haven't tested). I've had a brief look, but haven't been able to figure out what's wrong so far.

```
======================================================================
FAIL: testEvolvingConstraintsInterior (solvetests.TwoDimensionalTest)
Similar, but for the interior point method.
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/Users/snorf/Desktop/pyglpk/tests/solvetests.py", line 156, in testEvolvingConstraintsInterior
    self.assertEqual('nofeas', lp.interior())
AssertionError: 'nofeas' != None

----------------------------------------------------------------------
```

**Comment by bradfordboyle** [comment]
Originally, `pyglpk` called the GLPK function `lpx_interior()` which has been deprecated in favor of the newer function `glp_interior()`. Buried in the source of GLPK is an internal version of `lpx_interior()` that calls `glp_interior()` and translates the return value. To quickly get `pyglpk` compiling, I copied this code into src/lpx.c:106--120. The problem is that `glp_interior()` returns 0 when the LP has been solved---from the GLPK manual

> This code does not necessarily mean that the solver has found optimal solution. It only means that the solution process was successful

This means that the expected return value for an infeasible LP using an interior point solver is `None`. Checking `lp.status` will then return 'nofeas'.

The first option would be to change the test to reflect that `None` is the expected return for an infeasible LP and check that `lp.status` is 'nofeas'. The other option would be move this logic into the function `LPX_solver_interior()` (src/lp.c:552--557) so that test as written passes.

**Comment by snorfalorpagus** [comment]
If we're aiming for 0.3(.1) to be fully API compatible with code written for the original pyglpk 0.3 we should do the latter (i.e. modify `lp.c` so that the existing test passes).

**Comment by bradfordboyle** [comment]
Fixed in e51a75b.

**Comment by snorfalorpagus** [comment]
:+1:
